maximbircu/devtools-library
# Implement discard changes dialog

Closes: #73


### What has been done
1. Implemented a discard dialog - in case the user will try to leave the configuration screen with unsaved changes the screen will not close and a discard dialog will be displayed instead. The dialog will be closed and the unsaved changes discarded in case the user will hit the OK button.
2. Updated some libraries including kotlinx-serialization and migrated to AGP 7
3. Update targetSdkVersion to 31

### Screenshots
Android | iOS
:-: | :-:
<img src="https://user-images.githubusercontent.com/12527390/128254499-391ed680-8a10-4a0b-b3e3-a6693dafa054.png" width=200>| Will be implemented later |

### How to test
U1
1. Open the sample app and open any dev tools configuration screen
2. Adjust any dev tool value
3. Don't hit the apply button and leave the screen by press the back button
4. Notice the discard dialog
5. Hit "CANCEL" or tap outside the dialog
6. Make sure the dialog closes and the configuration screen remains open

U2
1. Open the sample app and open any dev tools configuration screen
2. Adjust any dev tool value
3. Don't hit the apply button and leave the screen by press the back button
4. Notice the discard dialog
5. Hit "OK" 
6. Make sure the dialog and the configuration screen closes and the unsaved changes are reset
